### DISCUSSION TITLE
How to create another apps folder

### DISCUSSION THREAD

**Discussion by ugo94490** [nasa/cFS#280] [Q&A] [answered]
Hello, I am currently working on a prototype using cFS and I have a question about the repository tree.

# Context

I would like to be able to create another apps folder where I could store all my applications that would be transversal to other projects.

This folder would be a submodule and would allow all my projects to take advantage of my previously created modules without copying and pasting the sources. Moreover I could update directly one of the transversal applications because the folder would be a submodule.

# Question

After some research I don't see how to add a specific folder to the cFS compilation, has anyone ever tried to compile applications located in another folder than apps?

**Comment by jphickey** [comment, marked as answer]
Yes, you can set the "MISSION_MODULE_SEARCH_PATH" variable in CMake.  This is a list of paths that are checked for apps.

For example I have something like this in my `global_build_options.cmake` file:

```
list(APPEND MISSION_MODULE_SEARCH_PATH
    "$ENV{HOME}/code/cfecfs/myapps"
)
```

**Reply by ugo94490** [reply]
Thank you very much for your help, I have succeeded in calling applications located in another folder.

I currently have a directory in this form:

* cFS-PoC:
	* cFS (submodule, github sources) 
	* cFS-common (submodule, transversal applications over multiple projects of our company)
		* Drv
			* SpaceWire
			* 1553
			* UART 
	* MissionSpecificApplications (custom applications for a specific project)
		* OperationalModeManagement
		* ImageProcessing
		* FDIR


I would like to know if it is possible to extract from cFS all the folders customizable to our specific project, like sample_defs or TC and TM located in the tools folder.

So that the cFS build chain will look for these configuration files specific to our project only in cFS-PoC and not in cFS

Is there a quick way to do it?Or do I have to modify all the paths one by one?

**Reply by skliper** [reply]
To be honest the expectation/assumption is for users to clone and own the cFS bundle (or assemble a mono-repo if that is preferred over submodule use) since it's really just intended to provide a sample "distribution" vs being a submodule within a real distribution.  So really just move what's in the cFS bundle up one layer which would avoid having to rearrange things too much:

* cFS-PoC:
	* cfe (submodule)
	* osal (submodule)
	* psp (submodule)
	* target_defs (your customized version) 
	* cFS-common (submodule, transversal applications over multiple projects of our company)
		* Drv
			* SpaceWire
			* 1553
			* UART 
	* MissionSpecificApplications (custom applications for a specific project)
		* OperationalModeManagement
		* ImageProcessing
		* FDIR
